### PR TITLE
Add documentation button

### PR DESCRIPTION
**Pull request by FilipZajdel**
Add a new optional property 'docsUrl' to the index schema that allows to set the documentation's url. In case it isn't used, the readme is automatically linked. If the readme does not exist, the repo url is linked.

Also add the link for Thingy 91X documentation.

VSC-2602
VSC-2776

image

### CODE REVIEW

File: site/src/schema.ts [hunk 1]
```diff
@@ -120,6 +120,10 @@ export const appMetadataSchema = {
             type: 'string',
             description: 'The default git branch that the repository is checked out. Inferred from the repo if missing.'
         },
+        docsUrl: {
+            type: 'string',
+            description: `The url of the addon's documentation`
```
Comment (greg-fer) [inline site/src/schema.ts:125]:
```suggestion
            description: `The URL of the add-on's documentation`
```

**Review by greg-fer — CHANGES_REQUESTED** [review]
Nice addition!
Will the button wording and icon change if it is linking to the add-on's repository instead of a README or docs? Having "Documentation" and its icon for the pure repo link might suggest the docs are missing by mistake, which might not be the case -- their lack could be intentional.

File: scripts/validate-index.ts [hunk 2]
```diff
@@ -33,6 +35,7 @@ async function run() {
         if (!isValid) {
             validate.errors?.forEach((error) => reportError(file, error));
             isError = true;
+            console.log(`${orgIndex} ${file}`)
```
Comment (kylebonnici) [inline scripts/validate-index.ts:38]: needed for testing?

**Review by kylebonnici — COMMENTED** [review]
Looks good to me. Will let @greg-fer also give his input as well

**Comment by FilipZajdel** [comment]
> Nice addition! Will the button wording and icon change if it is linking to the add-on's repository instead of a README or docs? Having "Documentation" and its icon for the pure repo link might suggest the docs are missing by mistake, which might not be the case -- their lack could be intentional.

I added the link to the repo as the last resort fallback in case there is neither docsUrl nor README. If that's confusing think I suggest to hiding the button completely. What do you think?

File: scripts/validate-index.ts (hunk 2 above)
Comment (FilipZajdel) [reply scripts/validate-index.ts:38]: Yup, a leftover after testing. Good catch!

**Comment by greg-fer** [comment]
> > Nice addition! Will the button wording and icon change if it is linking to the add-on's repository instead of a README or docs? Having "Documentation" and its icon for the pure repo link might suggest the docs are missing by mistake, which might not be the case -- their lack could be intentional.
> 
> I added the link to the repo as the last resort fallback in case there is neither docsUrl nor README. If that's confusing I suggest hiding the button completely. What do you think?

Sounds good!

**Review by greg-fer — APPROVED** [review]

**Pull request merged**